main.py

```diff
@@ -183,7 +186,7 @@ def main():
                 raise ValueError("No eos_token or bos_token found")
         tokenizer.pad_token = tokenizer.eos_token
         
-        args.generations_path = "generations.json" if args.generations_path is None else args.generations_path
+        generations_path = "generations.json" if args.generations_path is None else args.generations_path
```

Comment:
Nice! I think we need to still assign it to args, since args.generations_path will be used in Evaluator. Maybe instead we can use args.generation_path below i.e. `with open(args.generations_path, "w") as fp:`

Reply:
Makes sense. Will fix it.